generated/summaries/data_types.md

```diff
@@ -5,16 +5,155 @@ This is the course header. This will be added on top of every page. Go to [DoDAO
+    * Unlike a tuple, every element of an array must have the same data type.
+    * Arrays in Rust have a fixed length, i.e. they are not dynamic in nature by default.
+    * Arrays are useful when you want your data allocated on the stack rather than the memory heap.
+    * For Example:- 
+      ```
+        fn main() {
+            let a: [i32; 5] = [1, 2, 3, 4, 5];
+        }
+      ```
+      Here, i32 is the type of each element. After the semicolon, the number 5 indicates the array contains five elements.
+    *  We can access elements of an array using indexing, like this:
+      ```
+        fn main() {
+            let a = [1, 2, 3, 4, 5];
+        
+            let first = a[0]; // outputs 1
+            let second = a[1]; // outputs 2
+        }
+      ```
+ 
  **Variables**        
-Some Summary of Variables   
-- Value type1
+- Variables and Mutability
+  * In Rust, variables are immutable in nature by default.
+  * When a variable is immutable, once a value is bound to a name, you can’t change that value
+  * To make the variables mutable, we use the keyword:- `mut`.
+  * For Example:-
+    ```
+      fn main() {
+          let mut x = 5;
+          println!("The value of x is: {x}");
+          x = 6;
+          println!("The value of x is: {x}");
+      }
+    ```
+- Constants
+  * Constants are values that are bound to a name and are not allowed to change, i.e. they are also immutable.
+  * Usage of keyword `mut` isn't allowed with constants.
+  * Constants are immutable by default, and also they’re always immutable.
+  * We declare constants using the `const` keyword instead of the `let` keyword, and the type of the value must be annotated.
+  * Constants can be declared in any scope, including the global scope, making them useful for values that many parts of code need to know about.
+  * Constants may be set only to a constant expression, not the result of a value that could only be computed at runtime.
+  * For Example:-
+    ```
+      const THREE_HOURS_IN_SECONDS: u32 = 60 * 60 * 3;
+    ```
  
- **Bit Manupilation**        
-Some Summary of Bit Manupilation   
-- Value type1
+ **Keywords**        
+- Strict Keywords
+  * These keywords can only be used in their correct contexts. They cannot be used as the names of: Items, Variables and function parameters, etc.
+  * For Example:- ` mut, break, else, continue, return, impl, etc. ` 
+
+- Reserved Variables
+  * These keywords aren't used yet, but they are reserved for future use. They have the same restrictions as strict keywords.
+  * For Example:- ` abstract, try, do, final, typeof, etc. `
+
+- Weak Keywords
+  * These keywords have special meaning only in certain contexts.
+  * For example, it is possible to declare a variable or method with the name `union`.
+ 
+ **Bit Manipulation**        
+- Bit Mnipulation is strictly done by rust compilers for storing data in variables, for two's compliment and many other purposes.
+ 
+ **Strings**        
+- String
```

Comment:
We should expand on String and explain it in detail as "Strings" are a very important topic

Reply:
Totally Agreed, I'll get to work